src/lib/class/specialized_string/Password/methods.cpp

```diff
@@ -0,0 +1,64 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   methods.cpp                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gle-mini <gle-mini@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/03/04 22:37:23 by jodufour          #+#    #+#             */
+/*   Updated: 2024/03/13 12:17:11 by gle-mini         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "abnf_components.hpp"
+#include "class/specialized_string/Password.hpp"
+
+#define MAXIMUM_LENGTH_FOR_PASSWORD 50
+
+inline static bool is_nolfcrspcl(char c)
+{
+	return (c >= 0x01 && c <= 0x09) || (c >= 0x0B && c <= 0x0C) || (c >= 0x0E && c <= 0x1F) || (c >= 0x21 && c <= 0x39)
+	    || (c >= 0x3B);
+}
+
+inline static bool is_nolfcrsp(char c)
+{
+	return (c >= 0x01 && c <= 0x09) || (c >= 0x0B && c <= 0x0C) || (c >= 0x0E && c <= 0x1F) || (c >= 0x21);
+}
+
+inline static bool is_nolfcr(char c) { return c != '\n' && c != '\r'; }
```

Comment:
Missing `const` qualifier for `c` + This way is C style, you'd better do like this:
```c++
return std::string("\r\n").find(c) == std::string::npos;
```